layer/layer_store.go

```diff
@@ -659,15 +659,19 @@ func (ls *layerStore) initMount(graphID, parent, mountLabel string, initFunc Mou
 	}
```

Comment:
If we need to remove on any failure, we could use a defer;

1.  add a `retErr` output variable

```go
func (ls *layerStore) initMount(graphID, parent, mountLabel string, initFunc MountInit, storageOpt map[string]string) (_ string, retErr error) {
```

2. define a defer here that attempts to remove on failure;

```go
	defer func() {
		if retErr != nil {
			// cleanup on failure
			if err := ls.driver.Remove(initID); err != nil {
				log.G(context.TODO()).WithFields(log.Fields{
					"initID": initID,
					"error":  err,
				}).Warnf("Removing init layer after failure: %v", retErr)
			}
		}
	}()
```